substrate/client/chain-spec/src/lib.rs

```diff
@@ -16,19 +16,229 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Substrate chain configurations.
-//!
-//! This crate contains structs and utilities to declare
-//! a runtime-specific configuration file (a.k.a chain spec).
+#![allow(rustdoc::private_intra_doc_links)]
+
+//! This crate includes structs and utilities for defining configuration files (known as chain
+//! specification) for both runtime and node.
+//!
+//! # Intro: Chain Specification
+//!
+//! The chain specification comprises parameters and settings that define the properties and an
+//! initial state of a chain. Users typically interact with the JSON representation of the chain
+//! spec. Internally, the chain spec is embodied by the [`GenericChainSpec`] struct, and specific
+//! properties can be accessed using the [`ChainSpec`] trait.
+//!
+//! In summary, although not restricted to, the primary role of the chain spec is to provide a list
+//! of well-known boot nodes for the blockchain network and the means for initializing the genesis
+//! storage. This initialization is necessary for creating a genesis block upon which subsequent
+//! blocks are built. When the node is launched for the first time, it reads the chain spec,
+//! initializes the genesis block, and establishes connections with the boot nodes.
+//!
+//! The JSON chain spec is divided into three main logical sections:
+//! - one section details general chain properties,
+//! - another explicitly or indirectly defines the genesis storage, which, in turn, determines the
+//!   genesis hash of the chain,
+//! - the third deals with the runtime code.
+//!
+//! The chain specification consists of the following fields:
+//!
+//! <table>
+//!   <thead>
+//!     <tr>
+//!       <th>Chain spec key</th>
+//!       <th>Description</th>
+//!     </tr>
+//!   </thead>
+//!   <tbody>
+//!     <tr>
+//!       <td>name</td>
+//!       <td>The human readable name of the chain.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>id</td>
+//!       <td>The id of the chain.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>chainType</td>
+//!       <td>The chain type of this chain
+//!           (refer to
+//!            <a href="enum.ChainType.html" title="enum sc_chain_spec::ChainType">
+//!              <code>ChainType</code>
+//!            </a>).
+//!       </td>
+//!     </tr>
+//!     <tr>
+//!       <td>bootNodes</td>
+//!       <td>A list of multi addresses that belong to boot nodes of the chain.</td>
```

Comment:
```suggestion
//!       <td>A list of [multi addresses](https://github.com/multiformats/multiaddr) that belong to boot nodes of the chain.</td>
```

Reply:
I also suggest to write multiaddr (same in telemetryEndpoints) instead of "multi address"